tests/test_e2e_integration.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+End-to-end integration tests between the local machine and the remote Orquestra
+platform.
+
+These test cases connect to the Orquestra platform. Prior authentication to the
+Orquestra platform is required for running them. The roundtrip time can be 1-2
+minutes for test cases including assertions on the output of the computation.
+"""
+import pytest
+import numpy as np
+import math
+import yaml
+import os
+
+import pennylane as qml
+from pennylane_orquestra import OrquestraDevice, QeQiskitDevice, QeIBMQDevice
+import pennylane_orquestra.gen_workflow as gw
+from pennylane_orquestra.cli_actions import qe_submit, workflow_details
+
+from conftest import (
+    qe_list_workflow,
+    backend_specs_default,
+    operator_string_default,
+    qasm_circuit_default,
+    resources_default,
+)
+
+qiskit_analytic_specs = '{"module_name": "qeqiskit.simulator", "function_name": "QiskitSimulator", "device_name": "qasm_simulator"}'
+qiskit_sampler_specs = '{"module_name": "qeqiskit.simulator", "function_name": "QiskitSimulator", "device_name": "qasm_simulator", "n_samples": 1000}'
+
+analytic_tol = 10e-10
+
+# The tolerance for sampling is expected to be higher
+tol = 10e-2
+
+class TestWorkflowSubmissionIntegration:
+    """Test that workflow generation works as expected."""
+
+    @pytest.mark.parametrize("resources", [None, resources_default])
+    def test_can_submit_and_query_workflow_details(self, resources, tmpdir):
+        """Test that filling in the workflow template for getting expectation
+        values can be submitted to Orquestra and workflow details can be queried."""
+        # Skip if not logged in to Orquestra
+        try_resp = qe_list_workflow()
+        need_login_msg = "token has expired, please log in again\n"
+
+        if need_login_msg in try_resp:
+            pytest.skip("Has not logged in to the Orquestra platform.")
+
+        backend_component = "qe-forest"
+        op = ['["[Z0]"]']
+        circuits = [qasm_circuit_default]
+
+        # Fill in workflow template
+        workflow = gw.gen_expval_workflow(
+            backend_component, backend_specs_default, circuits, op, resources=resources
+        )
+        file_name = tmpdir.join("test_workflow.yaml")
+
+        with open(file_name, "w") as file:
+            d = yaml.dump(workflow, file)
+
+        # Submit a workflow
+        workflow_id = qe_submit(file_name)
```

Comment:
Just to clarify: this creates a workflow, stores it in a new file called "test_workflow.yaml" and submits that file, right? Probably not an issue, but should we remove that file after the test is finished?

Reply:
Good point! `keep_file=False` is the default for `qe_submit`, so the file is removed.